mindsdb/integrations/handlers/weaviate_handler/weaviate_handler.py

```diff
@@ -0,0 +1,402 @@
+from collections import OrderedDict
+from datetime import datetime
+from typing import List, Optional
+
+import weaviate
+import pandas as pd
+
+from mindsdb.integrations.libs.const import HANDLER_CONNECTION_ARG_TYPE as ARG_TYPE
+from mindsdb.integrations.libs.response import RESPONSE_TYPE
+from mindsdb.integrations.libs.response import HandlerResponse
+from mindsdb.integrations.libs.response import HandlerResponse as Response
+from mindsdb.integrations.libs.response import HandlerStatusResponse as StatusResponse
+from mindsdb.integrations.libs.vectordatabase_handler import (
+    FilterCondition,
+    FilterOperator,
+    TableField,
+    VectorStoreHandler,
+)
+from mindsdb.utilities import log
+
+
+class WeaviateDBHandler(VectorStoreHandler):
+    """This handler handles connection and execution of the Weaviate statements."""
+
+    name = "weaviate"
+
+    def __init__(self, name: str, **kwargs):
+        super().__init__(name)
+
+        self._connection_data = kwargs.get("connection_data")
+
+        self._client_config = {
+            "weaviate_url": self._connection_data.get("weaviate_url"),
+            "weaviate_api_key": self._connection_data.get(
+                "weaviate_api_key"
+            ),
+            "persist_directory": self._connection_data.get("persist_directory"),
+        }
+
+        # either host + port or persist_directory is required
+        # but not both
+        if (
+                self._client_config["weaviate_url"] is None
+                or self._client_config["weaviate_api_key"] is None
+        ):
+            raise Exception(
+                "Either url + auth client secret is required for weaviate connection!"
+            )
+
+        self._client = None
+        self.is_connected = False
```

Comment:
It would be much better with an annotation
self.is_connected: bool = False

Reply:
resolved in 2d930990be47f593eff6be05211d6c34e2395614